src/rgw/store/dbstore/common/dbstore.cc

```diff
@@ -395,17 +396,10 @@ int DB::remove_user(const DoutPrefixProvider *dpp,
   RGWObjVersionTracker objv_tracker = {};
 
   orig_info.user_id = uinfo.user_id;
-  ret = get_user(dpp, string("user_id"), "", orig_info, nullptr, &objv_tracker);
+  ret = get_user(dpp, string("user_id"), uinfo.user_id.id, orig_info, nullptr, &objv_tracker);
 
-  if (!ret && objv_tracker.read_version.ver) {
-    /* already exists. */
-
-    if (pobjv && (pobjv->read_version.ver != objv_tracker.read_version.ver)) {
-      /* Object version mismatch.. return ECANCELED */
-      ret = -ECANCELED;
```

Comment:
Since rados was not checking for it, I dropped it for dbstore also as 'pobjv' was not set and thus there was always mismatch in remove() op. 
But this seems to be because of a bug in RGWUser:init(). Fixed it now. Please review.